x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -1116,12 +1116,23 @@ private static void renderTypesTable(List<String> argNames) throws IOException {
             [%header.monospaced.styled,format=dsv,separator=|]
             |===
             """ + header + "\n" + table.stream().collect(Collectors.joining("\n")) + "\n|===\n";
-        LogManager.getLogger(getTestClass()).info("Writing function types:\n{}", rendered);
+        LogManager.getLogger(getTestClass()).info("Writing function types for [{}]:\n{}", functionName(), rendered);
         writeToTempDir("types", rendered, "asciidoc");
     }
 
     private static String functionName() {
-        return StringUtils.camelCaseToUnderscore(getTestClass().getSimpleName().replace("Tests", "")).toLowerCase(Locale.ROOT);
+        return mapFunctionName(
+            StringUtils.camelCaseToUnderscore(getTestClass().getSimpleName().replace("Tests", "")).toLowerCase(Locale.ROOT)
+        );
+    }
+
+    private static String mapFunctionName(String name) {
```

Comment:
This is the simplest change to achieve a fix, but presumably there could be a more generic change with some annotation to the test class, or test rule, or reflection lookup of a class method. For example, perhaps classes that wish to override this simply define `MyTests::functionName` and the existing static functionName method first tries to get that before falling back to automatic name mapping.

Reply:
I do like the idea of annotating something - though it looks like it'd have to be the test class.

Reply:
OK. I've changed it to use an annotation. That is certainly more generic.